orocos_kdl/src/chainiksolvervel_pinv_nso.cpp

```diff
@@ -121,18 +123,20 @@ namespace KDL
             A += qd*qd * weights(i)*weights(i);
         }
 
-        // Calculate inverse Jacobian Jc^-1
-        for (i = 0; i < nj; ++i) {
-            tmp(i) = weights(i)*(q_in(i) - opt_pos(i)) / A;
-        }
+        if (A > 1e-9) {
```

Comment:
As with most magic numbers, I don't have a strong reason, but 1e-9 seemed to work well during initial testing. At that value, the difference from the goal joint configuration, must be very small.